framework/db/BaseActiveRecord.php

```diff
@@ -418,7 +418,13 @@ public function getRelatedRecords()
      */
     public function hasAttribute($name)
     {
-        return isset($this->_attributes[$name]) || in_array($name, $this->attributes());
+        static $attributes = [];
+
+        if (!array_key_exists($name, $attributes)) {
+            $attributes[$name] = isset($this->_attributes[$name]) || in_array($name, $this->attributes());
+        }
+
+        return $attributes[$name];
```

Comment:
Right, it's technically possible, regardless of whether it makes sense to do so.

For these cases, perhaps we could place this logic in a separate class, document it and write about it in the Performance Tuning guide. (Or even in a separate extension if it's not wanted in the core)

```php
    public function attributes()
    {
        static $attributes = null;

        if ($attributes === null) {
            $attributes = parent::attributes();
        }

        return $attributes;
    }
```